### ISSUE TITLE
File Path is longer than 260 characters

### ISSUE THREAD

**Issue by DanceParty** [fractal-code/meteor-azure#9]
After running `meteor npm --version` I get npm version `3.10.9`.

```
Command: bash .config/azure/deploy.sh
The package.json file does not specify node.js engine version constraints.
The node.js application will run with the default node.js version 6.9.1.
Selected npm version 3.10.8
meteor-azure: Setting NPM version
D:\local\AppData\npm\npm -> D:\local\AppData\npm\node_modules\npm\bin\npm-cli.js
D:\local\AppData\npm
`-- npm@3.10.8 

meteor-azure: Building app
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'
npm WARN repository No description
npm WARN repository No repository field.
npm WARN repository No README data
npm WARN repository No license field.

Changes to your project's package version selections:

momentjs:moment  added, version 2.10.6

Handling node.js deployment.
KuduSync.NET from: 'D:\local\LocalAppData\meteor-azure\bundle' to: 'D:\home\site\wwwroot'
Copying file: '.node_version.txt'
Copying file: 'main.js'
Copying file: 'README'
Copying file: 'star.json'
Copying file: 'programs\server\boot-utils.js'
Copying file: 'programs\server\boot.js'
Copying file: 'programs\server\config.json'
Copying file: 'programs\server\mini-files.js'
Copying file: 'programs\server\npm-rebuild-args.js'
Copying file: 'programs\server\npm-rebuild.js'
Copying file: 'programs\server\npm-rebuilds.json'
Copying file: 'programs\server\npm-require.js'
Copying file: 'programs\server\npm-shrinkwrap.json'
Copying file: 'programs\server\package.json'
Copying file: 'programs\server\profile.js'
Copying file: 'programs\server\program.json'
Copying file: 'programs\server\server-json.js'
Copying file: 'programs\server\app\app.js'
Copying file: 'programs\server\app\app.js.map'
Copying file: 'programs\server\assets\app\equip-report.html'
Copying file: 'programs\server\assets\app\equip-report.js'
Copying file: 'programs\server\assets\app\layout.css'
Copying file: 'programs\server\assets\app\layout.html'
Copying file: 'programs\server\assets\app\layout.js'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.browser.html'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.cordova.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_popup_response.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_redirect_response.html'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.node_version'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.npm-shrinkwrap.json'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\acorn'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\babylon'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\detect-indent'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\jsesc'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\json5'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\loose-envify'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\mkdirp'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\regjsparser'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\repeating'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\user-home'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.editorconfig'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.gitattributes'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.meteor-portable'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.npmignore'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.tern-project'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.travis.yml'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\AUTHORS'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\CHANGELOG.md'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\LICENSE'
Omitting next output lines...
Processed 1628 files...
Processed 3532 files...
Processed 5364 files...
Error: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
An error has occurred during web site deployment.
Kudu Sync failed
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'\r\nnpm WARN repository No description\r\nnpm WARN repository No repository field.\r\nnpm WARN repository No README data\r\nnpm WARN repository No license field.\r\nError: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.\r\nD:\Program Files (x86)\SiteExtensions\Kudu\59.51102.2512\bin\Scripts\starter.cmd bash .config/azure/deploy.sh
```

**Comment by ramijarrar** [comment]
Can you confirm the version in your `.meteor/release`?

**Comment by DanceParty** [comment]
npm version? The only npm in `npm-mongo@2.2.11_2`.

**Comment by ramijarrar** [comment]
No, your Meteor version (in `.meteor/release`).

**Comment by DanceParty** [comment]
sorry, I misread. It is `METEOR@1.4.2`

**Comment by ramijarrar** [comment]
Do you have a package.json - can you paste the contents of that? (I will try to setup a repro)

**Comment by DanceParty** [comment]
The package.json in the root of my project only contains:

```
{
    "webshot": "0.16.0"
}
```

**Comment by ramijarrar** [comment]
Okay - I'll take a closer look tonight (if it isn't sensitive code, would be helpful if you could provide access to the repo)

**Comment by DanceParty** [comment]
Unfortunately, the repo is not owned by me, and it is in fact a private repo. I could provide more information though if need be!

**Comment by ramijarrar** [comment]
Spent some time debugging this and I've identified the cause to be a regression introduced in v1.1.0 which is leading to an incorrect internal NPM version. 

I'm working on a fix at the moment.

**Comment by ramijarrar** [comment]
This should be resolved in v1.1.1 (feel free to re-open if not the case)

**Comment by DanceParty** [comment]
Thanks I'll test it tonight! I appreciate the very quick response

**Comment by DanceParty** [comment]
I am still getting the same error :/

```
Command: bash .config/azure/deploy.sh
The package.json file does not specify node.js engine version constraints.
The node.js application will run with the default node.js version 6.9.1.
Selected npm version 3.10.8
meteor-azure: Setting NPM version
D:\local\AppData\npm\npm -> D:\local\AppData\npm\node_modules\npm\bin\npm-cli.js
D:\local\AppData\npm
`-- npm@3.10.8 

3.10.8
meteor-azure: Building app
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'
npm WARN repository No description
npm WARN repository No repository field.
npm WARN repository No README data
npm WARN repository No license field.
Handling node.js deployment.
Cannot removed the 'to be deleted' directory, ignoring
KuduSync.NET from: 'D:\local\LocalAppData\meteor-azure\bundle' to: 'D:\home\site\wwwroot'
Copying file: '.node_version.txt'
Copying file: 'main.js'
Copying file: 'README'
Copying file: 'star.json'
Copying file: 'programs\server\boot-utils.js'
Copying file: 'programs\server\boot.js'
Copying file: 'programs\server\config.json'
Copying file: 'programs\server\mini-files.js'
Copying file: 'programs\server\npm-rebuild-args.js'
Copying file: 'programs\server\npm-rebuild.js'
Copying file: 'programs\server\npm-rebuilds.json'
Copying file: 'programs\server\npm-require.js'
Copying file: 'programs\server\npm-shrinkwrap.json'
Copying file: 'programs\server\package.json'
Copying file: 'programs\server\profile.js'
Copying file: 'programs\server\program.json'
Copying file: 'programs\server\server-json.js'
Copying file: 'programs\server\app\app.js'
Copying file: 'programs\server\app\app.js.map'
Copying file: 'programs\server\assets\app\equip-report.html'
Copying file: 'programs\server\assets\app\equip-report.js'
Copying file: 'programs\server\assets\app\layout.css'
Copying file: 'programs\server\assets\app\layout.html'
Copying file: 'programs\server\assets\app\layout.js'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.browser.html'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.cordova.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_popup_response.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_redirect_response.html'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.node_version'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.npm-shrinkwrap.json'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\acorn'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\babylon'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\detect-indent'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\jsesc'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\json5'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\loose-envify'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\mkdirp'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\regjsparser'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\repeating'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\user-home'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.editorconfig'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.gitattributes'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.meteor-portable'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.npmignore'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.tern-project'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.travis.yml'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\AUTHORS'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\CHANGELOG.md'
Omitting next output lines...
Processed 1954 files...
Processed 3839 files...
Processed 5010 files...
Error: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
An error has occurred during web site deployment.
Kudu Sync failed
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'\r\nnpm WARN repository No description\r\nnpm WARN repository No repository field.\r\nnpm WARN repository No README data\r\nnpm WARN repository No license field.\r\nError: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.\r\nD:\Program Files (x86)\SiteExtensions\Kudu\59.51102.2512\bin\Scripts\starter.cmd bash .config/azure/deploy.sh
```

**Comment by ramijarrar** [comment]
Can you paste your application settings?

**Comment by DanceParty** [comment]
```
.Net Framework: v4.6
PHP Version: Off
Java Version: Off
Python Version: 2.7

Platform: 32-Bit
Web-Sockets: On
Always On: Off
Managed Pipeline Version: Integrated

ARR Affinity: On
Auto Swap: Off

WEBSITE_NODE_DEFAULT_VERSION: 6.9.1
MONGO_URL: <mongodblinkfromMongoLab>
ROOT_URL: https://websitename.azurewebsites.net
METEOR_SETTINGS: <insert Meteor Settings>
SCM_COMMAND_IDLE_TIMEOUT: 3600
METEOR_AZURE_NPM_VERSION: 3.10.8
```

**Comment by ramijarrar** [comment]
Try changing the following settings and then rebuild (this will also clear the cache):

```
WEBSITE_NODE_DEFAULT_VERSION: 4.6.1
METEOR_AZURE_NPM_VERSION: 3.10.9
```

**Comment by DanceParty** [comment]
I believe it is the same error... :(

```


Command: bash .config/azure/deploy.sh
The package.json file does not specify node.js engine version constraints.
The node.js application will run with the default node.js version 4.6.1.
Selected npm version 2.15.9
meteor-azure: Setting NPM version
D:\local\AppData\npm\npm -> D:\local\AppData\npm\node_modules\npm\bin\npm-cli.js
npm@3.10.9 D:\local\AppData\npm\node_modules\npm
3.10.9
meteor-azure: Building app
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'
npm WARN repository No description
npm WARN repository No repository field.
npm WARN repository No README data
npm WARN repository No license field.
Handling node.js deployment.
Cannot removed the 'to be deleted' directory, ignoring
KuduSync.NET from: 'D:\local\LocalAppData\meteor-azure\bundle' to: 'D:\home\site\wwwroot'
Copying file: '.node_version.txt'
Copying file: 'main.js'
Copying file: 'README'
Copying file: 'star.json'
Copying file: 'programs\server\boot-utils.js'
Copying file: 'programs\server\boot.js'
Copying file: 'programs\server\config.json'
Copying file: 'programs\server\mini-files.js'
Copying file: 'programs\server\npm-rebuild-args.js'
Copying file: 'programs\server\npm-rebuild.js'
Copying file: 'programs\server\npm-rebuilds.json'
Copying file: 'programs\server\npm-require.js'
Copying file: 'programs\server\npm-shrinkwrap.json'
Copying file: 'programs\server\package.json'
Copying file: 'programs\server\profile.js'
Copying file: 'programs\server\program.json'
Copying file: 'programs\server\server-json.js'
Copying file: 'programs\server\app\app.js'
Copying file: 'programs\server\app\app.js.map'
Copying file: 'programs\server\assets\app\equip-report.html'
Copying file: 'programs\server\assets\app\equip-report.js'
Copying file: 'programs\server\assets\app\layout.css'
Copying file: 'programs\server\assets\app\layout.html'
Copying file: 'programs\server\assets\app\layout.js'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.browser.html'
Copying file: 'programs\server\assets\packages\boilerplate-generator\boilerplate_web.cordova.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_popup_response.html'
Copying file: 'programs\server\assets\packages\oauth\end_of_redirect_response.html'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.node_version'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.npm-shrinkwrap.json'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\acorn'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\babylon'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\detect-indent'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\jsesc'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\json5'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\loose-envify'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\mkdirp'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\regjsparser'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\repeating'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\.bin\user-home'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.editorconfig'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.gitattributes'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.meteor-portable'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.npmignore'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.tern-project'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\.travis.yml'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\AUTHORS'
Copying file: 'programs\server\npm\node_modules\meteor\babel-compiler\node_modules\acorn\CHANGELOG.md'
Omitting next output lines...
Processed 2166 files...
Processed 4103 files...
Processed 5195 files...
Error: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
An error has occurred during web site deployment.
Kudu Sync failed
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'\r\nnpm WARN repository No description\r\nnpm WARN repository No repository field.\r\nnpm WARN repository No README data\r\nnpm WARN repository No license field.\r\nError: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.\r\nD:\Program Files (x86)\SiteExtensions\Kudu\59.51109.2534\bin\Scripts\starter.cmd bash .config/azure/deploy.sh
```

**Comment by ramijarrar** [comment]
I have a few ideas about what might be causing this but the only way to be sure is by identifying the offending line.

Can you try running again with the content of deploy.sh as per [this gist](https://gist.github.com/ramijarrar/1486411ee9ff00855c7e16df561759eb) (provides more verbose output).

**Comment by DanceParty** [comment]
Here is the last part of that error

```
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\lib\_stream_writable.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\core-util-is\float.patch'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\core-util-is\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\core-util-is\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\core-util-is\util.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\core-util-is\lib\util.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\inherits.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\inherits_browser.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\LICENSE'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\inherits\test.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\isarray\component.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\isarray\index.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\isarray\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\isarray\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\isarray\build\build.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\string_decoder\.npmignore'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\string_decoder\index.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\string_decoder\LICENSE'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\string_decoder\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\node_modules\readable-stream\node_modules\string_decoder\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\test\basic-test.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\test\sauce.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\bl\test\test.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\caseless\index.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\caseless\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\caseless\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\caseless\test.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\forever-agent\index.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\forever-agent\LICENSE'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\forever-agent\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\forever-agent\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\License'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\Readme.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\lib\form_data.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\.travis.yml'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\component.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\LICENSE'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\README.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\async\lib\async.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\License'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\Readme.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\lib\combined_stream.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\.gitignore'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\License'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\Makefile'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\package.json'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\Readme.md'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\lib\delayed_stream.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\test\common.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\test\run.js'
Copying file: 'programs\server\npm\node_modules\meteor\dfischer_phantomjs\node_modules\phantomjs\node_modules\request\node_modules\form-data\node_modules\combined-stream\node_modules\delayed-stream\test\integration\test-delayed-http-upload.js'
Error: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
An error has occurred during web site deployment.
Kudu Sync failed
npm WARN enoent ENOENT: no such file or directory, open 'D:\home\site\repository\package.json'\r\nnpm WARN repository No description\r\nnpm WARN repository No repository field.\r\nnpm WARN repository No README data\r\nnpm WARN repository No license field.\r\nError: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.\r\nD:\Program Files (x86)\SiteExtensions\Kudu\59.51109.2534\bin\Scripts\starter.cmd bash .config/azure/deploy.sh
```

**Comment by DanceParty** [comment]
If it is phantomjs that is causing the problem, I can actually redeploy without that package (we are not using it anymore)

**Comment by ramijarrar** [comment]
No -  it's definitely clear that the dependency tree is still not being flattened properly & since you have confirmed Meteor is at 1.4.x that means my original patch https://github.com/talos-code/meteor-azure/commit/a9d044bc6ed98164c6173041a2cb9ed84f6df514 didn't work properly.

I'm looking into this right now, should be able to push a new version today which resolves this.

**Comment by DanceParty** [comment]
Alright cool, I had just tried to remove phantomjs and it failed with another package. Thanks for the help!

**Comment by ramijarrar** [comment]
So it turns out the issue is specific to legacy atmosphere packages with dependencies that were shrink-wrapped with NPM 2.x and hence are not flattened properly (e.g [dfischer:phantomjs](https://raw.githubusercontent.com/dfischer/meteor-phantomjs/master/.npm/package/npm-shrinkwrap.json)) - I'm going to write some specific logic to handle these using a more resilient sync tool.

**Comment by ramijarrar** [comment]
Just released v1.1.2 which should finally solve this.

I actually ended up replacing the entire sync mechanism to be much more resilient against any of these issues (in case something similar pops up again in a different context).

Let me know when you get a chance to test it.

**Comment by DanceParty** [comment]
This worked perfectly. Thank you!
